src/parser.ts

```diff
@@ -140,6 +140,18 @@ export function extractFileFormat(format: string): BaseFileFormat {
   }
 }
 
+/**
+ * Checks that all values in a VariableDeclaration that depend on other values
+ * have those dependencies fulfilled, or throws an error if a value's dependency
+ * is missing.
+ * @param declaration a complete VariableDeclaration
+ */
+export function checkDependentValues(declaration: VariableDeclaration) {
+  if (declaration.format === 'file(json)' && !declaration.contentKey) {
```

Comment:
I don't want to hold up this review too much more. But im a little concerned with this only tackling `file(json)` If we add more file types in the future, this won't be effective.

In the lines above this, we check on the file format matching against `validFileFormats`. I can't help but wonder if we should be abstracting this pattern out ever so slightly to help here